### PR TITLE
new package: libandroid-execinfo

### LINKED ISSUE #10365: [Package]: libandroid-execinfo
### Package description

Backport of `<execinfo.h>` functionality (`backtrace{,_symbols{,_fd}}`) available since Android API 33.

### Home page URL

https://www.gnu.org/software/libc/manual/html_node/Backtraces.html

### Source code URL

https://android.googlesource.com/platform/bionic/+/refs/heads/master/libc/bionic/execinfo.cpp

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

As per request from @licy183. We already have `libexecinfo` package but reportedly it does not function on `x86_64`.

### PR DESCRIPTION
**Pull request by licy183**
Closes #10365

### CODE REVIEW

**Review by Grimler91 — COMMENTED** [review]
Maybe add a comment in the build.sh about which repo the files are taken from, to make it easier to check for updates upstream in the future

**Comment by buttaface** [comment]
Also, put something in the description so that users can figure out which libexecinfo package to use, feel free to change the other one's description too.

File: packages/libandroid-execinfo/execinfo.c [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+#include <unwind.h>
+
+static inline int __tmpfile_crate() {
+  char name[] = _PATH_TMP ".backtrace-FFFFFFFF-XXXXXXXX";
+  snprintf(name, sizeof(name), _PATH_TMP ".backtrace-%0x-XXXXXXXX", getpid() & 0xFFFFFFFF);
+  // XXX: It is almost impossible to have file conflicts, isn't it?
+  int fd = mkstemp(name);
+  unlink(name);
+  return fd;
+}
```
Comment (xtkoba) [inline packages/libandroid-execinfo/execinfo.c:48]: Try `memfd_create` syscall first. Define `memfd_create` as follows:

```c
#include <sys/syscall.h>

/* ... */

#if defined __ANDROID__ && __ANDROID_API__ < 30
#define memfd_create(name,flags) syscall(SYS_memfd_create,name,flags)
#endif
```

Fall back to `mkstemp` only when it failed (returned `-1`).

**Review by xtkoba — CHANGES_REQUESTED** [review]
Otherwise (than `memfd_create`) I see that the changes made to the original code are a matter of porting from C++ to C and nothing is strange. Thanks!

**Review by xtkoba — APPROVED** [review]

**Comment by buttaface** [comment]
We now have two packages that install a `libexecinfo.so`, one that describes itself as `A quick-n-dirty BSD licensed clone of backtrace facility found in the GNU libc` and this one as `Shared library for the backtrace system function`, so nobody will know which one to use. This is a mess.

**Comment by licy183** [comment]
> We now have two packages that install a `libexecinfo.so`, one that describes itself as `A quick-n-dirty BSD licensed clone of backtrace facility found in the GNU libc` and this one as `Shared library for the backtrace system function`, so nobody will know which one to use. This is a mess.

Yeah, it seems a bad thing. In my opinion, `libexecinfo` provides a much faster `backtrace` functionality than the bionic libc's port because it doens't use the file io, but it only works on arm arch. I failed to execute the binary linked against it under a x86_64 Android emulator.

**Pull request merged**